lib/src/modules/main_screen/bloc/main_screen_bloc.dart

```diff
@@ -1,53 +1,66 @@
-import 'package:autoequal/autoequal.dart';
-import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:template/src/repositories/user_repository/user_repository.dart';
 
-part 'main_screen_bloc.g.dart';
 part 'main_screen_event.dart';
 part 'main_screen_state.dart';
+part 'main_screen_bloc.freezed.dart';
 
 class MainScreenBloc extends Bloc<MainScreenEvent, MainScreenState> {
-  final UserRepository userRepository;
-
-  MainScreenBloc({required this.userRepository}) : super(InitState()) {
+  MainScreenBloc({required this.userRepository})
+      : super(const MainScreenState.initial()) {
     on<InitEvent>(_init);
     on<AddUserEvent>(_saveUser);
     on<RemoveUserEvent>(_removeUser);
     on<ReportSentryError>(_handleReportSentryError);
   }
+  final UserRepository userRepository;
 
   Future<void> _init(InitEvent event, Emitter<MainScreenState> emit) async {
-    final user = await userRepository.getUser('userKey');
-    emit(MainScreenState(user: user));
+    emit(const MainScreenState.loading());
+    try {
+      final user = await userRepository.getUser('userKey');
+      if (user != null) {
+        emit(MainScreenState.loaded(user));
+      } else {
+        emit(const MainScreenState.initial());
+      }
+    } catch (e) {
+      emit(const MainScreenState.error('Błąd pobierania użytkownika'));
+    }
   }
 
   Future<void> _saveUser(
     AddUserEvent event,
     Emitter<MainScreenState> emit,
   ) async {
-    final user = User(pk: 1, firstName: 'Jan', lastName: 'Nowak');
-    await userRepository.saveUser('userKey', user);
-    emit(MainScreenState(user: user));
+    emit(const MainScreenState.loading());
+    try {
+      final user = User(pk: 1, firstName: 'Jan', lastName: 'Nowak');
+      await userRepository.saveUser('userKey', user);
+      emit(MainScreenState.loaded(user));
+    } catch (e) {
+      emit(const MainScreenState.error('Błąd zapisywania użytkownika'));
+    }
```

Comment:
Given we're doing a breaking change via the db swap anyway, maybe it's also a good time to introduce a `Result` type, so bloc's doesn't have (and potentially forget) to `try/catch` errors from repositories. We can use e.g. (`result_type`)[https://pub.dev/packages/result_type].